daemon/daemon.go

```diff
@@ -1062,7 +1062,7 @@ func (daemon *Daemon) PullOnBuild(name string, authConfigs map[string]types.Auth
 		pullRegistryAuth = &resolvedConfig
 	}
 
-	if err := daemon.PullImage(ref, nil, pullRegistryAuth, output); err != nil {
+	if err := daemon.PullImage(context.TODO(), ref, nil, pullRegistryAuth, output); err != nil { //TODO(mgoelzer):  replace context.TODO()
```

Comment:
I think we should use `context.Background()` if we're not going to handle the user agent in this case.